include/remill/Arch/Runtime/Intrinsics.h

```diff
@@ -303,4 +305,111 @@ __remill_write_io_port_16(Memory *, addr_t, uint16_t);
 [[gnu::used, gnu::const]] extern Memory *
 __remill_write_io_port_32(Memory *, addr_t, uint32_t);
 
+// More specific hyper calls.
+[[gnu::used, gnu::const]] extern Memory *__remill_x86_set_segment_es(Memory *);
```

Comment:
Set to what? Should there be a `uint16_t` param here? If changing the segment selector, maybe an output should be a `addr_t &` to overwrite the `*_BASE` reg value.

Reply:
Same thing here. I don't think the information is there in the hyper call.

https://github.com/lifting-bits/remill/blob/master/lib/Arch/X86/Semantics/DATAXFER.cpp#L664-L669

I think we can refactor it but, given that this information is already missing from the hyper call, can we follow up with this later?

Reply:
Yes we can follow-this up later. I think a later follow-up should perhaps get rid of the hypercall, and instead use this directly in the semantics.